test/server/admin/stats_handler_test.cc

```diff
@@ -167,6 +183,309 @@ TEST_P(AdminStatsTest, HandlerStatsPlainText) {
+            "B300000(1,1) B600000(1,1) B1.8e+06(1,1) B3.6e+06(1,1)\n",
+            code_response.second);
+}
+
+TEST_P(AdminStatsTest, HandlerStatsPlainTextHistogramBucketsDisjoint) {
+  const std::string url = "/stats?histogram_buckets=disjoint";
+
+  Stats::Histogram& h1 = store_->histogramFromString("h1", Stats::Histogram::Unit::Unspecified);
+  Stats::Histogram& h2 = store_->histogramFromString("h2", Stats::Histogram::Unit::Unspecified);
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 200));
+  h1.recordValue(200);
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 300));
+  h1.recordValue(300);
+
+  store_->mergeHistograms([]() -> void {});
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 300));
+  h1.recordValue(300);
+
+  store_->mergeHistograms([]() -> void {});
+
+  CodeResponse code_response = handlerStats(url);
+  EXPECT_EQ(Http::Code::OK, code_response.first);
+  EXPECT_EQ("h1: B0.5(0,0) B1(0,0) B5(0,0) B10(0,0) B25(0,0) B50(0,0) B100(0,0) B250(0,1) "
+            "B500(1,2) B1000(0,0) B2500(0,0) B5000(0,0) B10000(0,0) B30000(0,0) B60000(0,0) "
+            "B300000(0,0) B600000(0,0) B1.8e+06(0,0) B3.6e+06(0,0)\nh2: No recorded values\n",
+            code_response.second);
+
+  code_response = handlerStats(url + "&usedonly");
+  EXPECT_EQ(Http::Code::OK, code_response.first);
+  EXPECT_EQ("h1: B0.5(0,0) B1(0,0) B5(0,0) B10(0,0) B25(0,0) B50(0,0) B100(0,0) B250(0,1) "
+            "B500(1,2) B1000(0,0) B2500(0,0) B5000(0,0) B10000(0,0) B30000(0,0) B60000(0,0) "
+            "B300000(0,0) B600000(0,0) B1.8e+06(0,0) B3.6e+06(0,0)\n",
+            code_response.second);
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h2), 300));
+  h2.recordValue(300);
+
+  store_->mergeHistograms([]() -> void {});
+
+  code_response = handlerStats(url + "&usedonly&filter=h2");
+  EXPECT_EQ(Http::Code::OK, code_response.first);
+  EXPECT_EQ("h2: B0.5(0,0) B1(0,0) B5(0,0) B10(0,0) B25(0,0) B50(0,0) B100(0,0) B250(0,0) "
+            "B500(1,1) B1000(0,0) B2500(0,0) B5000(0,0) B10000(0,0) B30000(0,0) B60000(0,0) "
+            "B300000(0,0) B600000(0,0) B1.8e+06(0,0) B3.6e+06(0,0)\n",
+            code_response.second);
+}
+
+TEST_P(AdminStatsTest, HandlerStatsPlainTextHistogramBucketsInvalid) {
+  const std::string url = "/stats?histogram_buckets=invalid_input";
+  CodeResponse code_response = handlerStats(url);
+  EXPECT_EQ(Http::Code::BadRequest, code_response.first);
+  EXPECT_EQ("usage: /stats?histogram_buckets=cumulative  or /stats?histogram_buckets=disjoint \n",
+            code_response.second);
+}
+
+TEST_P(AdminStatsTest, HandlerStatsJsonNoHistograms) {
+  const std::string url = "/stats?format=json&usedonly";
+
+  Stats::Counter& c1 = store_->counterFromString("c1");
+
+  c1.add(10);
+
+  store_->histogramFromString("h1", Stats::Histogram::Unit::Unspecified);
+
+  store_->mergeHistograms([]() -> void {});
+
+  const std::string expected_json = R"EOF({
+    "stats": [
```

Comment:
nit: this may be more readable as a single line string without the indent.

Reply:
Changed to single line